src/EditorFeatures/CSharpTest/AddParameter/AddParameterTests.cs

```diff
@@ -584,5 +589,1762 @@ void M()
+interface I1
+{
+    void M1(int v);
+}
+class C1 : I1
+{
+    void I1.M1() { }
+    void M2()
+    {
+        ((I1)this).M1(1);
+    }
+}";
+            var fix_All = @"
+interface I1
+{
+    void M1(int v);
+}
+class C1 : I1
+{
+    void I1.M1(int v) { }
+    void M2()
+    {
+        ((I1)this).M1(1);
+    }
+}";
+            await TestInRegularAndScriptAsync(code, fix_DeclarationOnly, index: 0);
+            await TestInRegularAndScriptAsync(code, fix_All, index: 1);
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocationImplicitInterface()
+        {
+            var code =
+@"
+interface I1
+{
+    void M1();
+}
+class C1 : I1
+{
+    public void M1() { }
+    void M2()
+    {
+        [|M1|](1);
+    }
+}";
+            var fix_DeclarationOnly = @"
+interface I1
+{
+    void M1();
+}
+class C1 : I1
+{
+    public void M1(int v) { }
+    void M2()
+    {
+        M1(1);
+    }
+}";
+            var fix_All = @"
+interface I1
+{
+    void M1(int v);
+}
+class C1 : I1
+{
+    public void M1(int v) { }
+    void M2()
+    {
+        M1(1);
+    }
+}";
+            await TestInRegularAndScriptAsync(code, fix_DeclarationOnly, index: 0);
+            await TestInRegularAndScriptAsync(code, fix_All, index: 1);
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        [Trait("TODO", "Fix broken")]
+        public async Task TestInvocationGenericMethod()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class C1
+{
+    void M1<T>(T arg) { }
+    void M2()
+    {
+        [|M1|](1, 2);
+    }
+}",
+@"
+class C1
+{
+    void M1<T>(T arg, int v) { }
+    void M2()
+    {
+        M1(1, 2);
+    }
+}");
+            //Should fix to: void M1<T>(T arg, T v) { }
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocationRecursion()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class C1
+{
+    void M1()
+    {
+        [|M1|](1);
+    }
+}",
+@"
+class C1
+{
+    void M1(int v)
+    {
+        M1(1);
+    }
+}");
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocationOverloads1()
+        {
+            var code =
+@"
+class C1
+{
+    void M1(string s) { }
+    void M1(int i) { }
+    void M2()
+    {
+        [|M1|](1, 2);
+    }
+}";
+            var fix0 =
+@"
+class C1
+{
+    void M1(string s) { }
+    void M1(int i, int v) { }
+    void M2()
+    {
+        M1(1, 2);
+    }
+}";
+            var fix1 =
+@"
```

Comment:
what was reasoning behind offering fix1? tagging @jinujoseph @kuhlenh sine this seems design question on the feature.